backends/ubpf/ubpfDeparser.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2019 Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "ubpfDeparser.h"
+#include "ubpfType.h"
+#include "frontends/p4/methodInstance.h"
+
+namespace UBPF {
+
+    UBPFDeparserTranslationVisitor::UBPFDeparserTranslationVisitor(
+            const UBPFDeparser *deparser) :
+            CodeGenInspector(deparser->program->refMap,
+                             deparser->program->typeMap), deparser(deparser),
+            p4lib(P4::P4CoreLibrary::instance) {
+        setName("UBPFDeparserTranslationVisitor");
+    }
+
+    void UBPFDeparserTranslationVisitor::compileEmitField(const IR::Expression *expr,
+                                                          cstring field,
+                                                          unsigned alignment,
+                                                          EBPF::EBPFType *type) {
+
+        unsigned widthToEmit = dynamic_cast<EBPF::IHasWidth *>(type)->widthInBits();
```

Comment:
Do you want to check whether the dynamic_cast can return NULL?

Reply:
I've added additional check.